persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -283,6 +296,8 @@ public static void inferRelationDetails(Module entityModule) {
                                             // one-to-many relation. associated entity is the owner.
                                             field.setRelation(computeRelation(entity, assocEntity, false));
                                             assocfield.setRelation(computeRelation(assocEntity, entity, true));
+                                        } else if (field.isArrayType() || field.getFieldType().equals("byte")) {
+                                            field.setRelation(null);
```

Comment:
Aren't the supported types defined as constants

Reply:
The issue here was, when the field is array type, it was classified as a relation type. This is not true for byte arrays. So we have to specifically set it to null here as null is the default value for non related fields